src/devices/src/virtio/net/device.rs

```diff
@@ -191,6 +178,29 @@ impl Net {
         })
     }
 
+    /// Create a new virtio network device with the given TAP interface.
+    pub fn new_with_tap(
```

Comment:
Maybe not related to this PR, but to me would make more sense that this is called `new` and the other one is called `new_with_tap` since you actually send a `Tap` as arg. WDYT?